src/DynamicNLPModels.jl

```diff
@@ -493,7 +504,8 @@ end
 
 function _build_condensed_blocks(
     s0, Q, R, A, B, N;
-    Qf = Q)
+    Qf = Q, 
+    S = zeros(size(Q, 1), size(R, 1)))
```

Comment:
consider initializing `S=nothing`. In that way, we can avoid unnecessarily allocating memory